mdaviz/chartview.py

```diff
@@ -117,3 +123,203 @@ def setLeftAxisUnits(self, text):
+            "pos1": None,
+            2: False,
+            "pos2": None,
+        }
+
+        # Plot configuration
+        config = {
+            "title": self.setPlotTitle,
+            "y": self.setLeftAxisText,
+            "x": self.setBottomAxisText,
+        }
+        for k, func in config.items():
+            func(kwargs.get(k))
+
+    def plot(self, *args, **kwargs):
+        # Plot data on the axes
+        self.main_axes.plot(*args, **kwargs)
+        self.main_axes.legend()
+        self.main_axes.grid(True, color="#cccccc", linestyle="-", linewidth=0.5)
+        self.update_info_panel()
+        self.canvas.draw()
+
+    def setPlotTitle(self, text):
+        self.main_axes.set_title(text, fontsize=FONTSIZE, y=1.03)
+
+    def setBottomAxisText(self, text):
+        self.main_axes.set_xlabel(text, fontsize=FONTSIZE, labelpad=10)
+
+    def setLeftAxisText(self, text):
+        self.main_axes.set_ylabel(text, fontsize=FONTSIZE, labelpad=20)
+
+    def clearPlot(self):
+        self.main_axes.clear()
+        self.cursors[1] = False
+        self.cursors[2] = False
+        self.info_panel_axes.clear()
+        self.canvas.draw()
+
+    # Additional methods:
+
+    # TODO: use a dictionary to track the cursors
+    # TODO: allow a blue cursor to be placed without the presence of a red cursor
+
+    def onclick(self, event):
+        # Check if the click was in the main_axes
+        if event.inaxes is self.main_axes:
+            # Middle click for red cursor
+            if event.button == 2:
+                if self.cursors[1]:
+                    self.cursors[1].remove()  # Remove existing red cursor
+                (self.cursors[1],) = self.main_axes.plot(
+                    event.xdata, event.ydata, "r+", markersize=15, linewidth=2
+                )
+                # Update cursor position
+                self.cursors["pos1"] = (event.xdata, event.ydata)
+
+            # Right click for blue cursor
+            elif event.button == 3:
+                if self.cursors[2]:
+                    self.cursors[2].remove()  # Remove existing blue cursor
+                (self.cursors[2],) = self.main_axes.plot(
+                    event.xdata, event.ydata, "b+", markersize=15, linewidth=2
+                )
+
+                # Update cursor position
+                self.cursors["pos2"] = (event.xdata, event.ydata)
+
+            # Update the info panel with cursor positions
+            self.update_info_panel()
+
+            # Redraw the canvas to display the new markers
+            self.canvas.draw()
+
+    def update_info_panel(self):
+        # Clear the info panel by removing each text artist
+        while len(self.info_panel_axes.texts) > 0:
+            for text in self.info_panel_axes.texts:
+                text.remove()
+
+        # Initialize text variables
+        cursor1_text = "Cursor 1: press middle click"
+        cursor2_text = "Cursor 2: press right click"
+        diff_text = "Difference: n/a"
```

Comment:
Similar, these keys could be in a dictionary that contains these computed results.